eva/executor/delete_executor.py

```diff
@@ -0,0 +1,33 @@
+from typing import Generator, Iterator
+
+from eva.executor.abstract_executor import AbstractExecutor
+from eva.executor.executor_utils import ExecutorError, apply_project
+from eva.models.storage.batch import Batch
+from eva.plan_nodes.project_plan import ProjectPlan
+from eva.utils.logging_manager import logger
+
+
+class DeleteExecutor(AbstractExecutor):
+    """ """
+
+    def __init__(self, node: ProjectPlan):
+        super().__init__(node)
+        self.target_list = node.target_list
+
+    def validate(self):
+        pass
+
+    def exec(self, **kwargs) -> Iterator[Batch]:
+        # Uses Where clause
+        try:
+            child_executor = self.children[0]
+            for batch in child_executor.exec(*args, **kwargs):
+                batch = apply_predicate(batch, self.predicate)
```

Comment:
Delete should not be yielding anything. It should rather call `delete` from the storage manager on the table. The predicate should also be based on the delete command.  https://github.com/georgia-tech-db/eva/blob/4b4ee7ac2e01e2eab0d7c17725b36ee011c410f9/eva/storage/sqlite_storage_engine.py#L189
Supporting arbitrary predicates (for example, containing UDFs like similarity ) won't be straightforward. In such cases, we can filter tuples ourselves and then call delete base don `_row_id` but we can think of it as an extension in later PR.

Reply:
Yes Gaurav, thanks for that
I have made changes accordingly.